packages/@lwc/module-resolver/src/index.ts

```diff
@@ -114,7 +148,71 @@ export function resolveModules(
     resolverConfig: Partial<ModuleResolverConfig> = {}
 ): RegistryEntry[] {
     const normalizedConfig = normalizeConfig(resolverConfig);
-    const rootConfig = loadConfig(normalizedConfig.rootDir);
-    const modules = mergeModules(normalizedConfig.modules, rootConfig.modules);
-    return resolveModulesFromList(modules, { root: normalizedConfig.rootDir });
+    const { rootDir, modules } = normalizedConfig;
+    const configPath = findFirstUpwardConfigPath(rootDir);
+
+    if (!configPath) {
+        throw new Error(`Unable to find a configurations to start the resolution`);
+    }
+
+    const { version } = loadPackageJson(configPath);
+    const lwcConfig = normalizeConfig({ ...getLwcConfig(configPath), rootDir: configPath });
+    const mergedModules = mergeModules(modules, lwcConfig.modules);
+    return resolveModulesFromList(mergedModules, {
+        rootDir,
+        scopeDir: rootDir,
+        skipExposeFilter: true,
+        version,
+    });
 }
+
+export function resolveModule(
+    importee: string,
+    importer: string,
+    ignoreCache = false
+): RegistryEntry | undefined {
+    if (
+        !importer ||
+        importee.startsWith('.') ||
+        importee.startsWith('/') ||
+        !fs.existsSync(importer)
```

Comment:
Yes it is neccesary. All of the playground (or anything that produce synthetic files) will flow up because the importer does not exists hence we don't know which root to use.